ui/src/app/components/object-detail/object-detail.component.ts

```diff
@@ -346,12 +346,60 @@ export class ObjectDetailComponent implements OnInit {
 
   dropColumn(element: any) {
     let colName = element.get('srcColName').value
-    this.spRowArray.value.forEach((col: IColumnTabData, i: number) => {
-      if (col.srcColName === colName) {
-        this.droppedColumns.push(col)
+    let associatedIndexes = this.getAssociatedIndexs(colName)
+    if (this.checkIfPkColumn(colName) || associatedIndexes.length != 0) {
```

Comment:
this check doesn't seem to work if we try dropping a column after modifying its name - it is allowing it to be dropped even though it has dependencies

Reply:
Fixed.